File: license-tracker-backend/src/main/java/com/prodapt/license_tracker_backend/service/VendorService.java
package com.prodapt.license_tracker_backend.service;


import com.prodapt.license_tracker_backend.dto.CreateVendorRequest;
import com.prodapt.license_tracker_backend.dto.UpdateVendorRequest;
import com.prodapt.license_tracker_backend.dto.VendorResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface VendorService {
    VendorResponse createVendor(CreateVendorRequest request);
    VendorResponse updateVendor(Long id, UpdateVendorRequest request);
    VendorResponse getVendorById(Long id);
    Page<VendorResponse> getAllVendors(Pageable pageable);
    List<VendorResponse> getAllVendorsList();
    void deleteVendor(Long id);
    boolean existsByVendorName(String vendorName);
}

File: licensetracker_frontend/src/services/vendorService.js
// src/services/vendorService.js
import api from './api';

const vendorService = {
    getAllVendors: async (page = 0, size = 10, sortBy = 'vendorName') => {
        const response = await api.get(`/vendors?page=${page}&size=${size}&sortBy=${sortBy}`);
        return response.data;
    },

    getAllVendorsList: async () => {
        const response = await api.get('/vendors/list');
        return response.data;
    },

    getVendorById: async (id) => {
        const response = await api.get(`/vendors/${id}`);
        return response.data;
    },

    createVendor: async (vendorData) => {
        const response = await api.post('/vendors', vendorData);
        return response.data;
    },

    updateVendor: async (id, vendorData) => {
        const response = await api.put(`/vendors/${id}`, vendorData);
        return response.data;
    },

    deleteVendor: async (id) => {
        const response = await api.delete(`/vendors/${id}`);
        return response.data;
    },

    checkVendorNameExists: async (vendorName) => {
        const response = await api.get(`/vendors/exists?vendorName=${vendorName}`);
        return response.data.exists;
    }
};

export default vendorService;
